src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -112,25 +152,17 @@ export class RuntimeLanguageContext {
         T extends codewhispererClient.ListRecommendationsRequest | codewhispererClient.GenerateRecommendationsRequest
     >(request: T): T {
         const fileContext = request.fileContext
-        const childLanguage = request.fileContext.programmingLanguage
-        let parentLanguage: codewhispererClient.ProgrammingLanguage
-        switch (childLanguage.languageName) {
-            case 'tsx':
-                parentLanguage = { languageName: CodeWhispererConstants.typescript }
-                break
-            case 'jsx':
-                parentLanguage = { languageName: CodeWhispererConstants.javascript }
-                break
-            default:
-                parentLanguage = childLanguage
-                break
+        const runtimeLanguage: codewhispererClient.ProgrammingLanguage = {
```

Comment:
This is nuts. How can there be so many names for the same thing in the same subystem?

- "programming language"
- "runtime language"

Use as few names as possible.

Reply:
yea agree, it's really crazy to have these different names in cw module.

codewhisperer language is a super set of runtime language(i.e. normalized language)
JSX / TSX will be normalized to JS/TS

ProgammingLanguage is the SDK POJO of the format { languageName: string }

does this more sense to have this?
`type NormalizedLanguageId = Exclude<CodewhispererLanguage, 'jsx' | 'tsx'>`
https://github.com/aws/aws-toolkit-vscode/pull/3940/files#diff-0509671083ef22655748504f51cc4ad12de62ff0ec7a4f317f57d6bf24652838R12

Reply:
the reason why I still keep the following 3 types existing in the module
1. type CodewhispererLanguage (which is telemetry.gen type)
2. type PlatformLanguageId
3. NormalizedLanguageId

is to enforce the compiler to do the string check at compile time